pkg/operator/ceph/cluster/mgr/spec.go

```diff
@@ -99,14 +103,6 @@ func (c *Cluster) makeDeployment(mgrConfig *mgrConfig) *apps.Deployment {
 		},
 	}
 	k8sutil.AddRookVersionLabelToDeployment(d)
-	if len(c.annotations) == 0 {
-		prometheusAnnotations := map[string]string{
-			"prometheus.io/scrape": "true",
-			"prometheus.io/port":   strconv.Itoa(metricsPort),
-		}
-		podSpec.ObjectMeta.Annotations = prometheusAnnotations
```

Comment:
This code sets the annotations at the same place you have added above, to the `podSpec`.
If the user has not given any annotations, these Prometheus annotations (line 104-105) are added by default.

Line 108 is setting it on the Deployment, which might be unnecessary but shouldn't cause any issues for Prometheus though.

Reply:
spec.template is not populated

```
spec:
  progressDeadlineSeconds: 600
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: rook-ceph-mgr
      ceph_daemon_id: a
      instance: a
      mgr: a
      rook_cluster: rook-ceph
  strategy:
    type: Recreate
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: rook-ceph-mgr
        ceph_daemon_id: a
        instance: a
        mgr: a
        rook_cluster: rook-ceph
      name: rook-ceph-mgr-a
```
while annotations are not set in a CephCluster